README.md

```diff
@@ -1,223 +1,300 @@
+Let's suppose that between TIMESTAMP4 and TIMESTAMP5 we modified the code to output a file `summary.txt`, but that otherwise nothing has changed. We would like to check that our file `output.csv` hasn't changed but oops! We've just overwritten it. Luckily we can compare to the json at TIMESTAMP4.
 
-## Randomness
-Hashing tells you whether something is the same, or different. It cannot tell you if something is almost the same. If your analysis is non-deterministic, you be getting a different hash every time. To deal with this, I recommend setting a random seed. Whatever language you're using should be able to provide you with documentation on how to do this.
+```
+catalogue compare \
+  catalogue_results/TIMESTAMP4.json \
+  catalogue_results/TIMESTAMP5.json
+```
 
-## Which hash function should we use?
-sha 512? I'm not an expert on the security of hash functions, is there any official guidance?
+Let us also suppose that the `metadata.json` output includes a timestamp. The diff we would expect would look something like this:
 
-## Ignoring files
-It's no help to hash files if we don't care about them changing. A good example of a file we wouldn't want to track is a `.git` file. We could consider a couple of ways to ignore files
-* Passing a `--ignore` flag to `catalogue`
-* Maintaining some kind of `.catalogueignore` config file within each directory just as you would for Git
-* automatically ignoring any `.` files
+```
+results differ in *3* places:
+=============================
+timestamp
+code
+results/latest_results/metadata.json
 
-## What happens if the hashes in .lock don't match the new hashes?
-I think we should record this discrepancy in the json file and inform the command line user.
+results matched in *2* places:
+==============================
+input_data
+results/latest_results/output.csv
 
-## Running `catalogue engage` and `catalogue disengage` in the wrong order
-`catalogue disengage` will check that a `.lock` file exists. If it doesn't it will warn
+results could not be compared in *1* places:
+============================================
+results/latest_results/summary.text
 ```
-Oops looks like you're already disengaged
-To engage, run "catalogue engage..."
+
+Of course this is what we *want*. We might find that our `output.csv` file *had* changed, and these hashes alone would do nothing to help us recover TIMESTAMP4 version, but they are enough to inform us of the problem, and importantly they do this without us having to permanently store the output of every analysis we run.
+
+### Share outputs
+
+We can then send a zip file of the results to a colleague along with the hash json produced during the final analysis (`TIMESTAMP5.json`).
+
+They can rerun the analysis and use `catalogue` to check that the json they received is the same as their state:
+
+```{bash}
+catalogue compare --hashes TIMESTAMP4.json
```

Comment:
The `--hashes`` option has been removed. Up to two hash jsons are just consumed as non-flagged standard input arguments. Also if only one json is provided then you need to set the input, code, and output flags, so I might add those here for clarity.